overlays/docker-compose.monitoring.yml

```diff
@@ -24,93 +24,98 @@ services:
   database:
     logging: *loki-logging
 
-  traefik:
-    logging: *loki-logging
-
-  # This is needed to avoid permission issues with the loki container when running as non-root user
-  loki-init:
-    image: busybox:1.37.0
-    user: root
+  grafana:
+    image: grafana/grafana:${GRAFANA_VERSION:-10.0.0}
     volumes:
-      - loki:/loki
-    command: ["sh", "-c", "mkdir -p /loki/chunks /loki/rules /loki/wal /loki/boltdb-shipper-compactor && chown -R nobody:nobody /loki"]
-
-  loki:
-    image: grafana/loki:${LOKI_VERSION:-2.9.0}
+      - grafana:/var/lib/grafana
+      - ./config/grafana/provisioning:/etc/grafana/provisioning:ro
+    environment:
+      GF_SECURITY_ADMIN_PASSWORD: ${GRAFANA_ADMIN_PASSWORD}
+      GF_SECURITY_ADMIN_USER: admin
+      GF_USERS_ALLOW_SIGN_UP: false
+      GF_SECURITY_DISABLE_GRAVATAR: true
+      GF_ANALYTICS_REPORTING_ENABLED: false
+      GF_ANALYTICS_CHECK_FOR_UPDATES: false
+      GF_LOG_LEVEL: warn
+      GF_PATHS_PROVISIONING: /etc/grafana/provisioning
+      GF_SERVER_ROOT_URL: https://grafana.${HOSTNAME}
+      GF_SERVER_SERVE_FROM_SUB_PATH: false
     restart: unless-stopped
-    user: nobody:nobody
-    security_opt:
-      - no-new-privileges:true
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:3000/api/health"]
+      <<: *healthcheck-options
+    user: grafana:grafana
     cap_drop:
       - ALL
+    read_only: true
+    security_opt:
+      - no-new-privileges:true
+
+  loki:
+    image: grafana/loki:${LOKI_VERSION:-2.9.0}
+    depends_on:
+      loki-init:
+        condition: service_completed_successfully
     volumes:
       - loki:/loki
-    # Port is exposed as the Loki Docker Driver is running on the host network and we need to send logs to it from the service containers
-    ports:
-      - "3100:3100"
     environment:
       LOKI_RETENTION_PERIOD: ${LOKI_RETENTION_PERIOD:-744h}
-    depends_on:
-      loki-init:
-        condition: service_completed_successfully
+    # Port is exposed as the Loki Docker Driver is running on the host network and we need to send logs to it from the service containers
+    ports:
+      - "127.0.0.1:3100:3100"
+    restart: unless-stopped
     healthcheck:
       test: ["CMD", "curl", "-f", "http://host.docker.internal:3100/ready"]
       <<: *healthcheck-options
-
-  prometheus:
-    image: prom/prometheus:${PROMETHEUS_VERSION:-v2.45.0}
-    restart: unless-stopped
     user: nobody:nobody
-    read_only: true
-    security_opt:
-      - no-new-privileges:true
     cap_drop:
       - ALL
+    security_opt:
+      - no-new-privileges:true
+
+  # This is needed to avoid permission issues with the loki container when running as non-root user
+  loki-init:
+    image: busybox:1.37.0
+    volumes:
+      - loki:/loki
+    command:
```

Comment:
Some of these changes are unrelated to the pre-commit but specific to the editor you're using. One of the main goals of this task is to have a common set of rules and tools which could ensure the project is formatted correctly regardless of which editor the individual team member is using